src/main/java/rx/schedulers/ExecutorScheduler.java

```diff
@@ -73,11 +66,15 @@ public Subscription schedule(Action0 action) {
             if (isUnsubscribed()) {
                 return Subscriptions.unsubscribed();
             }
-            ExecutorAction ea = new ExecutorAction(action, tasks);
+            ScheduledAction ea = new ScheduledAction(action, tasks);
             tasks.add(ea);
             queue.offer(ea);
             if (wip.getAndIncrement() == 0) {
                 try {
+                    // note that since we schedule the emission of potentially multiple tasks
+                    // there is no clear way to cancel this schedule from individual tasks
+                    // so even if executor is an ExecutorService, we can't associate the future
+                    // returned by submit() with any particular ScheduledAction
```

Comment:
This comment is somewhat confusing to me. The "potentially multiple tasks" is just that we're scheduling the "this" reference over and over again isn't it?

We are scheduling a single "this" one at a time so that it then drains from the queue to ensure sequential execution. 

Thus, the scheduling on the `Executor` has little to do with any particular `Action0`. Is that what you're saying?